www/source/2017-07-14-Habitat-mysql-cluster.html.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Making a Leader/Follower MySql Cluster with Habitat!
+date: 2017-07-14
+author: Nell Shamrell-Harrington
+tags: supervisor, packaging
+category: Supervisor
+classes: body-article
+---
+
+One of the most compelling pieces of Habitat is using the supervisor to create self-organizing and self-healing topologies.  Today we will, step by step, create a MySQL Leader/Follower cluster using Habitat.
+
+## Setting up your infrastructure
+
+Create three virtual machines on the cloud provider of your choice (when creating this post, I used AWS EC2).
+
+Why virtual machines and not containers?  You certainly could create a cluster in containers, but I strongly prefer to use containers only for stateless portions of an application.  A database is stateful part of an application and should be in a long lasting form of infrastructure, rather than an ephemeral one.
```

Comment:
This is a great point here and our blog really needs it! I might add that containers give you isolation, immutability, and the ability to schedule work loads. Since databases aren't schedulable, Habitat is actually a perfect choice because you get 2 of the 3 super powers that containers give you